test/integration/order.test.ts

```diff
@@ -77,4 +81,38 @@ describe('order', () => {
       owner: selfAlias,
     })
   })
+
+  it('creates an order transaction on chain', async () => {
+    demandCreateMock()
+    const { body: { id: orderId } } = await post(app, '/order', { parametersAttachmentId })
+
+    // submit to chain
+    const response = await post(app, `/order/${orderId}/creation`, {})
+    expect(response.status).to.equal(201)
+
+    const { id: transactionId, state } = response.body
+    expect(transactionId).to.match(
+      /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$/
+    )
+    expect(state).to.equal(TransactionState.submitted)
+
+    // check local transaction updates
+    const [transaction] = await db.getTransaction(transactionId)
+    expect(transaction.state).to.equal(TransactionState.finalised)
+    expect(transaction).to.contain({
+      state: 'finalised',
+      apiType: 'order',
+      transactionType: 'creation',
```

Comment:
checking `state: finalised` twice? Also can use the enums (soon to be types!) for equality check

Reply:
I'm not going to use enums for values, sorry but I can not allow myself to do that.

Reply:
you're using enum as value with `expect(transaction.state).to.equal(TransactionState.finalised)` so why the inconsistency?

Reply:
Um I think the point is ln 101 is literally contained within the assertion on ln 102. That said this is a nothing burger and this shouldn't prevent this merging.

@n3op2 I'm not sure what you're proposing here other than making a general gripe? If there's a point to make then fine but that's not how this is coming across

Reply:
@mattdean-digicatapult not sure if I understood you. I just found this picky.

Reply:
@mattdean-digicatapult yes, line has been removed, no need to double assert:) - useless